File: apps/ios/Barns/BarnsApp/Domain/UseCases/MyItems/ArchiveProductItemUseCase.swift
import Foundation

/// Archives a registered greenery locally by setting its status to `.archived`.
/// This is a soft action — the item is kept in the local store (no hard
/// delete) and all official content links (categoryId, careGuideIds, imageUrl)
/// and other fields are preserved. Local-only: never sent to a server.
struct ArchiveProductItemUseCase {
    private let repository: ProductItemRepository

    init(repository: ProductItemRepository) {
        self.repository = repository
    }

    func execute(_ item: ProductItem) async throws -> ProductItem {
        let archived = ProductItem(
            id: item.id,
            categoryId: item.categoryId,
            name: item.name,
            type: item.type,
            installedOrPurchasedAt: item.installedOrPurchasedAt,
            locationLabel: item.locationLabel,
            status: .archived,
            careGuideIds: item.careGuideIds,
            notes: item.notes,
            imageUrl: item.imageUrl,
            updatedAt: Date()
        )
        try await repository.updateProductItem(archived)
        return archived
    }
}

File: apps/android/app/src/main/java/com/barns/app/domain/usecase/myitems/ArchiveProductItemUseCase.kt
package com.barns.app.domain.usecase.myitems

import com.barns.app.domain.model.ProductItem
import com.barns.app.domain.model.ProductItemStatus
import com.barns.app.domain.repository.ProductItemRepository
import java.time.Instant

/**
 * Archives a registered greenery locally by setting its status to ARCHIVED.
 * This is a soft action: the item stays in the local store, official content
 * links are preserved, and nothing is sent to a server.
 */
class ArchiveProductItemUseCase(
    private val repository: ProductItemRepository,
) {
    suspend fun execute(item: ProductItem): ProductItem {
        val archived = item.copy(
            status = ProductItemStatus.ARCHIVED,
            updatedAt = Instant.now(),
        )
        repository.updateProductItem(archived)
        return archived
    }
}
